ui/components/src/StatChart/StatChart.tsx

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useMemo } from 'react';
+import { Box, Typography } from '@mui/material';
+import { merge } from 'lodash-es';
+import { use, EChartsCoreOption } from 'echarts/core';
+import { GaugeChart as EChartsGaugeChart, GaugeSeriesOption } from 'echarts/charts';
+import { LineChart as EChartsLineChart, LineSeriesOption } from 'echarts/charts';
+import { GridComponent, DatasetComponent, TitleComponent, TooltipComponent } from 'echarts/components';
+import { CanvasRenderer } from 'echarts/renderers';
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+import { formatValue, UnitOptions } from '../model/units';
+import { EChart } from '../EChart';
+import { GraphSeries } from '../model/graph';
+
+use([
+  EChartsGaugeChart,
+  EChartsLineChart,
+  GridComponent,
+  DatasetComponent,
+  TitleComponent,
+  TooltipComponent,
+  CanvasRenderer,
+]);
+
+const PANEL_PADDING = 32;
+const MIN_VALUE_SIZE = 12;
+const MAX_VALUE_SIZE = 36;
+
+export interface StatChartData {
+  calculatedValue: number | null | undefined;
+  seriesData: GraphSeries | null | undefined;
```

Comment:
Nit: Any time I see both `null` and `undefined`, I feel like something is a little off. 😂  Do they both serve a purpose here? If not, we've tried to stick to just `undefined` rather than also including `null`. Just makes for simpler code when you don't have to check for both.